daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cloudflare-workerskv.md

```diff
@@ -0,0 +1,242 @@
+  # Note that they will both be set to the same value.
+  binding = ""
+  id = ""
+  ```
+  
+  > Note: see the next section for how to generate an Ed25519 key pair. Make sure you use the **public** part of the key when deploying a Worker!
+
+4. Copy the (pre-compiled and minified) code of the Worker in the `worker.js` file. You can do that with this command:  
+  
+  ```sh
+  # Set this to the version of Dapr that you're using
+  DAPR_VERSION="release-{{% dapr-latest-version short="true" %}}"
+  curl -LfO "https://raw.githubusercontent.com/dapr/components-contrib/${DAPR_VERSION}/internal/component/cloudflare/workers/code/worker.js"
+  ```
+
+5. Deploy the Worker using Wrangler:
+  
+  ```sh
+  npx wrangler publish
+  ```
+
+Once your Worker has been deployed, you will need to initialize the component with these two metadata options:
+
+- **`workerName`**: Name of the Worker script. This is the value you set in the `name` property in the `wrangler.toml` file.
+- **`workerUrl`**: URL of the deployed Worker. The `npx wrangler command` will show the full URL to you, for example `https://mydaprkv.mydomain.workers.dev`.
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+## Generate an Ed25519 key pair
+
+All Cloudflare Workers listen on the public Internet, so Dapr needs to use additional authentication and data protection measures to ensure that no other person or application can communicate with your Worker (and thus, with your Worker KV namespace). These include industry-standard measures such as:
+
+- All requests made by Dapr to the Worker are authenticated via a bearer token (technically, a JWT) which is signed with an Ed25519 key. 
+- All communications between Dapr and your Worker happen over an encrypted connection, using TLS (HTTPS).
+- The bearer token is generated on each request and is valid for a brief period of time only (currently, one minute).
+
+To let Dapr issue bearer tokens, and have your Worker validate them, you will need to generate a new Ed25519 key pair. Here are examples of generating the key pair using OpenSSL or the step CLI.
+
+{{< tabs "Generate with OpenSSL" "Generate with the step CLI" >}}
+
+{{% codetab %}}
+<!-- Generate with OpenSSL -->
+
+> Support for generating Ed25519 keys is available since OpenSSL 1.1.0, so the commands below will not work if you're using an older version of OpenSSL.
+
+> Note for Mac users: on macOS, the "openssl" binary that is shipped by Apple is actually based on LibreSSL, which as of writing doesn't support Ed25519 keys. If you're using macOS, either use the step CLI, or install OpenSSL 3.0 from Homebrew using `brew install openssl@3` then replacing `openssl` in the commands below with `$(brew --prefix)/opt/openssl@3/bin/openssl`.
+
+You can generate a new Ed25519 key pair with OpenSSL using:
+
+```sh
+openssl genpkey -algorithm ed25519 -out private.pem
+openssl pkey -in private.pem -pubout -out public.pem
+
+# On Mac, using openssl@3 from Homebrew:
```

Comment:
Same here. Ok to split into its own code block?

Reply:
Done